openmc/deplete/nuclide.py

```diff
@@ -0,0 +1,570 @@
+                self._yield_data = fission_yields
+            else:
+                self._yield_data = FissionYieldDistribution(fission_yields)
+
+    @property
+    def yield_energies(self):
+        if self._yield_data is None:
+            return None
+        return self.yield_data.energies
+
+    @classmethod
+    def from_xml(cls, element, fission_q=None):
+        """Read nuclide from an XML element.
+
+        Parameters
+        ----------
+        element : xml.etree.ElementTree.Element
+            XML element to write nuclide data to
+        fission_q : None or float
+            User-supplied fission Q value [eV].
+            Will be read from the element if not given
+
+        Returns
+        -------
+        nuc : openmc.deplete.Nuclide
+            Instance of a nuclide
+
+        """
+        nuc = cls()
+        nuc.name = element.get('name')
+
+        # Check for half-life
+        if 'half_life' in element.attrib:
+            nuc.half_life = float(element.get('half_life'))
+            nuc.decay_energy = float(element.get('decay_energy', '0'))
+
+        # Check for decay paths
+        for decay_elem in element.iter('decay'):
+            d_type = decay_elem.get('type')
+            target = decay_elem.get('target')
+            branching_ratio = float(decay_elem.get('branching_ratio'))
+            nuc.decay_modes.append(DecayTuple(d_type, target, branching_ratio))
+
+        # Check for reaction paths
+        for reaction_elem in element.iter('reaction'):
+            r_type = reaction_elem.get('type')
+            Q = float(reaction_elem.get('Q', '0'))
+            branching_ratio = float(reaction_elem.get('branching_ratio', '1'))
+
+            # If the type is not fission, get target and Q value, otherwise
+            # just set null values
+            if r_type != 'fission':
+                target = reaction_elem.get('target')
+            else:
+                target = None
+                if fission_q is not None:
+                    Q = fission_q
+
+            # Append reaction
+            nuc.reactions.append(ReactionTuple(
+                r_type, target, Q, branching_ratio))
+
+        fpy_elem = element.find('neutron_fission_yields')
+        if fpy_elem is not None:
+            nuc.yield_data = FissionYieldDistribution.from_xml_element(fpy_elem)
+
+        return nuc
+
+    def to_xml_element(self):
+        """Write nuclide to XML element.
+
+        Returns
+        -------
+        elem : xml.etree.ElementTree.Element
+            XML element to write nuclide data to
+
+        """
+        elem = ET.Element('nuclide')
+        elem.set('name', self.name)
+
+        if self.half_life is not None:
+            elem.set('half_life', str(self.half_life))
+            elem.set('decay_modes', str(len(self.decay_modes)))
+            elem.set('decay_energy', str(self.decay_energy))
+            for mode, daughter, br in self.decay_modes:
```

Comment:
Nit: If `mode, daughter, br` is deconstructing a `DecayTuple` type, I believe that would mean that `mode` maps to `DecayTuple.type`.  If this is the case (and intended behavior), I would prefer renaming `mode` to `mode_type` for clarity.  This would also help on line 228.

Note that I definitely do not recommend renaming `mode` to `type`, as `type` is a overloaded term in python.